src/main/scala/edg_ide/runner/PDFGeneratorUtil.scala

```diff
@@ -1,32 +1,76 @@
 package edg_ide.runner
 
-import com.lowagie.text.{Document, Rectangle}
+import com.lowagie.text.{Document, Element, HeaderFooter, Rectangle}
 import com.lowagie.text.pdf.PdfWriter
 import edg_ide.swing.ElkNodePainter
+import edgir.elem.elem.{BlockLike, HierarchyBlock}
 import org.eclipse.elk.graph.ElkNode
+import edg.util.SeqMapSortableFrom._
+import edg.wir.{DesignPath, ProtoUtil}
+import edg_ide.edgir_graph.HierarchyGraphElk
 
 import java.awt.Color
-import java.io.FileOutputStream
+import java.io.{FileNotFoundException, FileOutputStream, IOException}
 
 
 object PDFGeneratorUtil{
 
-  def generate(rootNode: ElkNode, fileName: String): Unit = {
-    // TODO: Set a fixed document size and scale graphics accordingly?
-    val width = rootNode.getWidth.toFloat + 2 * ElkNodePainter.margin.toFloat
-    val height = rootNode.getHeight.toFloat + 2 * ElkNodePainter.margin.toFloat
-    val document = new Document(new Rectangle(width, height))
+  private def generatePageSize(node: ElkNode): (Float, Float) = {
+    val width = node.getWidth.toFloat + 2 * ElkNodePainter.margin.toFloat
+    val height = node.getHeight.toFloat + 2 * ElkNodePainter.margin.toFloat
+    (width, height)
+  }
+
+  def generate(content: HierarchyBlock, fileName: String): Unit = {
+    try {
+      val document = new Document()
+      val writer = PdfWriter.getInstance(document, new FileOutputStream(fileName))
+      val header = new HeaderFooter(true)
+      header.setBorder(Rectangle.NO_BORDER)
+      header.setAlignment(Element.ALIGN_RIGHT)
+      document.setFooter(header)
+
+      def printNode(node: ElkNode): Unit = {
+        val (width, height) = generatePageSize(node)
+        document.setPageSize(new Rectangle(width, height))
+
+        if (document.isOpen) {
+          document.newPage
+        }
+        else {
```

Comment:
```suggestion
        } else {
```

General style preference here is this more compact style